src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -198,10 +219,34 @@ class EthService(
     val totalDifficulty = uncleHeaderOpt.flatMap(uncleHeader => blockchain.getTotalDifficultyByHash(uncleHeader.hash))
 
     //The block in the response will not have any txs or uncles
-    val uncleBlockResponseOpt = uncleHeaderOpt.map { uncleHeader => BlockResponse(blockHeader = uncleHeader, totalDifficulty = totalDifficulty) }
+    val uncleBlockResponseOpt = uncleHeaderOpt.map { uncleHeader =>
+      BlockResponse(blockHeader = uncleHeader, totalDifficulty = totalDifficulty, pendingBlock = false) }
     Right(UncleByBlockHashAndIndexResponse(uncleBlockResponseOpt))
   }
 
+  /**
+    * Implements the eth_getUncleByBlockNumberAndIndex method that fetches an uncle from a certain index in a requested block.
+    *
+    * @param request with the number/tag of the block and the index of the uncle requested
+    * @return the uncle that the block has at the given index or None if the client doesn't have the block or if there's no uncle in that index
+    */
+  def getUncleByBlockNumberAndIndex(request: UncleByBlockNumberAndIndexRequest): ServiceResponse[UncleByBlockNumberAndIndexResponse] = Future {
+    val UncleByBlockNumberAndIndexRequest(blockParam, uncleIndex) = request
+    val uncleHeaderOpt = resolveBlock(blockParam).toOption
+      .flatMap { block =>
+        if (uncleIndex >= 0 && uncleIndex < block.body.uncleNodesList.size)
+          Some(block.body.uncleNodesList.apply(uncleIndex.toInt))
+        else
+          None
+      }
+    val totalDifficulty = uncleHeaderOpt.flatMap(uncleHeader => blockchain.getTotalDifficultyByHash(uncleHeader.hash))
+
+    //The block in the response will not have any txs or uncles
+    val uncleBlockResponseOpt = uncleHeaderOpt.map { uncleHeader =>
+      BlockResponse(blockHeader = uncleHeader, totalDifficulty = totalDifficulty, pendingBlock = blockParam == BlockParam.Pending) }
```

Comment:
maybe `resolveBlock` should return information if block is pending
maybe something like
`val (uncleHeaderOpt, pending) = resolveBlock(blockParam)` ?